src/server/appserver.cpp

```diff
@@ -350,6 +351,25 @@ AppServer::AppServer(int &argc, char **argv)
         }
     }
 
+    appStateValue = "";
+    if (bool found = false; !ParmsDb::instance()->selectAppState(AppStateKey::LogUploadToken, appStateValue, found) || !found) {
+        LOG_ERROR(_logger, "Error in ParmsDb::selectAppState");
+    }
+    const std::string logUploadToken = std::get<std::string>(appStateValue);
+    if (!logUploadToken.empty()) {
+        UploadSessionCancelJob cancelJob(UploadSessionType::LogUpload, logUploadToken);
+        const ExitCode exitCode = cancelJob.runSynchronously();
+        if (exitCode != ExitCodeOk) {
+            LOG_WARN(_logger, "Error in UploadSessionCancelJob::runSynchronously : " << exitCode);
+        } else {
+            LOG_INFO(_logger, "Previous Log upload api call cancelled");
+            if (bool found = false;
+                !ParmsDb::instance()->updateAppState(AppStateKey::LogUploadToken, std::string(), found) || !found) {
```

Comment:
```suggestion
                !ParmsDb::instance()->updateAppState(AppStateKey::LogUploadToken, AppStateValue{""}), found) || !found) {
```

Reply:
Not sure to see the benefice of this approach?